### DISCUSSION TITLE
Security, App Tracking, Data tied to the user or org.

### DISCUSSION THREAD

**Discussion by FONickReichard** [kienstra/localci#256] [General]
Hello local-ci team,

This looks like an awesome tool. I've been looking for ways to test pipelines locally to speed up the dev flow. I am, however, committed to adhering to my company's stringent privacy policy, which restricts the download of extensions that haven't been officially approved. 

Before I approach my employer to request a review of this extension to our approved list, it would be a big help if you could provide me with more detailed information regarding its privacy features. I reviewed the information available here: https://marketplace.visualstudio.com/items?itemName=LocalCI.local-ci, and noticed that it mentions, "Local CI has no server that runs jobs, so the site has no knowledge of the jobs or any data from them." This is certainly reassuring.

I do have a few follow-up questions related to privacy:

Aside from telemetry, does the extension collect or transmit any user data?
Does it access or manipulate any application data?
Are there any interactions with external/internal services or data sources under the free account?

I'll spend some time to look at the source. Wanted to ask in the mean-time for others that may have similar questions.

Any feedback would be great! 

Best

**Comment by kienstra** [comment]
Hi @FONickReichard,
Thanks a lot for considering Local CI.

>Aside from telemetry, does the extension collect or transmit any user data?

No, other than an email that it asks for and you can decline.

It will [ask for your email](https://github.com/getlocalci/local-ci/pull/131#issue-1253150146) the first time you use the extension, but you can simply click outside the box and ignore the prompt. 

If you decide to enter an email, it will [post the email to getlocalci.com](https://github.com/getlocalci/local-ci/pull/131/files#diff-9047accdbc8d1b51b39c0f882cb0649f34cce1ba6de06c4fa8981b2b0d09ce80R9), but won't store it in VS Code, or on your machine.

>Does it access or manipulate any application data?

No

It [creates](https://github.com/search?q=repo%3Agetlocalci%2Flocal-ci%20tmp&type=code) a `/tmp/local-ci/` directory to store logs and configs from previous jobs. 

But I don't think that's application data.

>Are there any interactions with external/internal services or data sources under the free account?

Yes, it [runs the](https://github.com/getlocalci/local-ci/blob/17f12a6ef5fcaaef329b7640e39ccd4f7aff2828/src/job/JobRunner.ts#L130) the config file through the Docker container [mikefarah/yq](https://hub.docker.com/r/mikefarah/yq) to create [cleaner terminal output](https://github.com/getlocalci/local-ci/pull/96). If this is a blocker for you, I could think about disabling it.

Local CI interacts with your local machine's Docker a lot, as it runs all of the CircleCI® jobs on Docker.

Though you probably knew that, as you can't use it without Docker.

**Comment by kienstra** [comment]
Sorry for the delay, and feel free to email me at ryan@getlocalci.com.
